### ISSUE TITLE
bug?: Using ts parser does not replace correctly

### ISSUE THREAD

**Issue by rxliuli** [facebook/jscodeshift#511]
I'm using Jscodeshift to replace the following code in astexplorer testing, but I get different results using babel/ts

```ts
export const systemCoreApi = wrapApiClient<ISystemApi>({  namespace: 'system' })
```

use babel parser

```ts
export const parser = "babel";

// Press ctrl+space for code completion
export default function transformer(file, api) {
  const j = api.jscodeshift;
  const root = j(file.source);

  root
    // .find(j.TSTypeReference)
    .find(j.Identifier, { name: "ISystemApi" })
    .replaceWith((path) => {
      path.node.name = "ISystemCoreApi";
      return path.node;
    });

  return root.toSource();
}
```

output

```ts
export const systemCoreApi = wrapApiClient<ISystemCoreApi>({  namespace: 'system' })
```

use ts parser

```ts
// jscodeshift can take a parser, like "babel", "babylon", "flow", "ts", or "tsx"
// Read more: https://github.com/facebook/jscodeshift#parser
export const parser = "ts";

// Press ctrl+space for code completion
export default function transformer(file, api) {
  const j = api.jscodeshift;
  const root = j(file.source);

  root
    // .find(j.TSTypeReference)
    .find(j.Identifier, { name: "ISystemApi" })
    .replaceWith((path) => {
      path.node.name = "ISystemCoreApi";
      return path.node;
    });

  return root.toSource();
}
```

output

```ts
export const systemCoreApi = wrapApiClient<ISystemApi>({  namespace: 'system' })
```

Lastly, I'm wondering how to use the @typescript-eslint/parser parser, it seems to work best for parsing ts in ast explorer (the result of ts parsing is very different from other parsers and jscodeshift's api)

![image](https://user-images.githubusercontent.com/24560368/179156237-f42df5c3-e051-498a-9985-035b7a5367c7.png)
![image](https://user-images.githubusercontent.com/24560368/179156289-50a4f175-875e-45da-8899-17b3e0eba9ea.png)

**Comment by Daniel15** [comment]
Hmm, if I remove the `export const parser = 'ts'` and select TypeScript from the dropdown list on ASTExplorer, it works fine? https://astexplorer.net/#/gist/a87887cd564a07395d7f2b1a3f336d2a/2d968bb30b8ca3fdda9b8a31645f1d7e69ca8248

**Comment by ElonVolo** [comment]
I’ve looked into this a bit. If you go parser/tsOptions and remove line 42, ‘typescript’ it works.

There’s an entry in the docs for babel-plugin-transform-typescript that mentions:

“2. This plugin dos not support export = and import =“

I suspect this has something to do with it.

**Comment by rxliuli** [comment]
> Hmm, if I remove the `export const parser = 'ts'` and select TypeScript from the dropdown list on ASTExplorer, it works fine?

Yes, but this doesn't actually use the ts parser, e.g. if you include an enum in it, it will error out

```ts
export const systemCoreApi = wrapApiClient<ISystemApi>({  namespace: 'system' })

export enum S {
 name = 1 
}
```

![image](https://user-images.githubusercontent.com/24560368/180904421-297cb05a-3ae5-4a52-bad8-f7fe2d72ce92.png)

**Comment by rxliuli** [comment]
> export const parser = "@typescript-eslint/parser";

Using it will give an error as if it's not used correctly, and I've also confirmed that the underlying recast is fine and can be used normally `@typescript-eslint/parser`

![image](https://user-images.githubusercontent.com/24560368/180904593-2cac79e4-7d4a-4357-9b4d-c5fda2dd67de.png)
